NARFO_BE/NARFO_BE/Controllers/AdministratorsController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using NARFO_BE.Models;
+
+namespace NARFO_BE.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AdministratorsController : ControllerBase
+    {
+        private readonly narfoContext _context;
+
+        public AdministratorsController(narfoContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/Administrators
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Administrator>>> GetAdministrator()
+        {
+            return await _context.Administrator.ToListAsync();
+        }
+
+        // GET: api/Administrators/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Administrator>> GetAdministrator(int id)
+        {
+            var administrator = await _context.Administrator.FindAsync(id);
+
+            if (administrator == null)
+            {
+                return NotFound();
+            }
+
+            return administrator;
+        }
+
+        // PUT: api/Administrators/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutAdministrator(int id, Administrator administrator)
+        {
+            if (id != administrator.AdminId)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(administrator).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!AdministratorExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
```

Comment:
Why are you just throwing the exception here?

Reply:
this auto generated code.... we'll change everything accordingly when we work on those particular controllers if that's ok